common/throttle_timer.go

```diff
@@ -12,54 +11,85 @@ If a long continuous burst of .Set() calls happens, ThrottleTimer fires
 at most once every "dur".
 */
 type ThrottleTimer struct {
-	Name string
-	Ch   chan struct{}
```

Comment:
lets make it unidirectional so it's clear

Reply:
Thank you. Very good.

Reply:
Not so fast...

```
./throttle_timer.go:50:10: invalid operation: close(t.Ch) (cannot close receive-only channel)
./throttle_timer.go:55:9: invalid operation: t.Ch <- struct {} literal (send to receive-only type <-chan struct {})
./throttle_timer_test.go:49:23: cannot use t.Ch (type <-chan struct {}) as type chan struct {} in field value
```

I think I need to expose it read only and keep a write copy as a private variable?

Reply:
yeah, this was a bad idea given current impl. If we want to make this right, we need to unexpose Ch and create a method >>

```
func (t ...) Ch() <-chan struct{} {
  return t.Ch
}

```

Reply:
Ah yes. The right way would be to make this private and have a method that returns the `<-chan`. But that would make this breaking. Why don't we expose that method and mark the `Ch` as deprecated, and make private at some future release. keeping around the `output` copy is weird, let's drop it

Reply:
I don't mind breaking changes